Atlas/frontend/src/containers/CulturalHeritage/Add.js

```diff
@@ -1,102 +1,182 @@
+                            headers: {'Authorization': 'JWT ' + token},
+                            data: {
+                                images: [{url: addCHInputs.img_url}]
+                            }
+                        }
+                    ).then(_resp => {
+                        addSuccess(dispatch);
+                    })
+                } else {
+                    addSuccess(dispatch);
+                }
+            }).catch(err => {
+                dispatch(addCHFail(err.response.data));
+            });
+        },
+        addCHTag: (name) => dispatch(addCHTag(name)),
+        updateGeo: (coords) => dispatch(updateLocation(coords)),
+        deleteCHTag: (id) => dispatch(deleteCHTag(id)),
+        searchLocation: (addCHInputs) => {
+
+            //geocoding (obtain lat and lng from given address)
+            axios({
+                method: 'post',
+                url: 'https://maps.googleapis.com/maps/api/geocode/json?address=' + addCHInputs.locationName + '&key=AIzaSyCJ-k7tMf86LbMbQ-YAwi6-YAGmTx1z064'
+            }).then(response => {
+
+
+                if (response.data.status == "OK") {
+                    const name1 = "lng";
+                    const val1 = JSON.stringify(response.data.results[0].geometry.location.lng);
+                    dispatch(updateCHInput(name1,Number(val1).toFixed(5)));
+                    const name2 = "lat";
+                    const val2 = JSON.stringify(response.data.results[0].geometry.location.lat);
+                    dispatch(updateCHInput(name2,Number(val2).toFixed(5)));
+
+
+                    const locationName = JSON.stringify(response.data.results[0].address_components[0].long_name);
+                    dispatch(updateCHInput("locationName",locationName));
+
+                    console.log(locationName);
+                    console.log(Number(val1).toFixed(5)+" and "+Number(val2).toFixed(5));
+                    console.log(addCHInputs.lat + " and " + addCHInputs.lng + addCHInputs.locationName);
+                } else {
+		
+		    console.log("wrong location"+addCHInputs.locationName);
+                   
+                }
+
+
+            })
+        },
+        mapClick: (mapProps, map, clickEvent) => {
+            dispatch(updateCHInput("lat", clickEvent.latLng.lat().toFixed(5)));
+            dispatch(updateCHInput("lng", clickEvent.latLng.lng().toFixed(5)));
+
+            //reverse-geocoding (obtain address from lat and lng)
+            axios({
+                method: 'post',
+                url: 'https://maps.googleapis.com/maps/api/geocode/json?latlng=' + clickEvent.latLng.lat().toFixed(5) + ',' + clickEvent.latLng.lng().toFixed(5) + '&key=AIzaSyCJ-k7tMf86LbMbQ-YAwi6-YAGmTx1z064'
+            }).then(response => {
+
+
+                if (response.data.status == "OK") {
+
+                    const locationName = JSON.stringify(response.data.results[0].address_components[1].long_name);
```

Comment:
Sometimes this part gives undefined error. Because it fails to retrieve an object with long_name. Just so you know